internal/trace/trace.go

```diff
@@ -107,5 +202,34 @@ func TracePrintf(ctx context.Context, attrMap map[string]interface{}, format str
 		}
 		attrs = append(attrs, a)
 	}
-	trace.FromContext(ctx).Annotatef(attrs, format, args...)
+	return attrs
+}
+
+// otAttrs converts a generic map to OpenTelemetry attributes.
+func otAttrs(attrMap map[string]interface{}) []attribute.KeyValue {
+	var attrs []attribute.KeyValue
+	// Sort the input map by its keys for predictable order in tests.
+	keys := make([]string, 0, len(attrMap))
+	for k := range attrMap {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
```

Comment:
Can we just move the sorting to the test as that is where it is needed for. Don't want to have a runtime cost for non-test code.

Reply:
Done, thanks!